### PR TITLE
MueLu: fix reuse

### PR DESCRIPTION
**Pull request by jhux2**
Fixes issue #7411, adds additional unit test.

## Testing
Added new test that demonstrates bug and bug fix.

---
This change is [Reviewable](https://reviewable.io/reviews/trilinos/trilinos/7433)

### CODE REVIEW

File: packages/muelu/src/Interface/MueLu_HierarchyManager.hpp [hunk 1]
```diff
@@ -249,7 +250,7 @@ namespace MueLu {
       if (!matvecParams_.is_null())
         H.SetMatvecParams(matvecParams_);
       // FIXME: Should allow specification of NumVectors on parameterlist
```
Comment (cgcgcg) [inline packages/muelu/src/Interface/MueLu_HierarchyManager.hpp:252]: I guess that comment doesn't apply anymore..

File: packages/muelu/src/Interface/MueLu_HierarchyManager.hpp [hunk 2]
```diff
@@ -93,6 +93,7 @@ namespace MueLu {
         doPRrebalance_          (MasterList::getDefault<bool>("repartition: rebalance P and R")),
         implicitTranspose_      (MasterList::getDefault<bool>("transpose: use implicit")),
         fuseProlongationAndUpdate_ (MasterList::getDefault<bool>("fuse prolongation and update")),
+        sizeOfMultiVectors_     (MasterList::getDefault<int>("number of vectors")),
```
Comment (cgcgcg) [inline packages/muelu/src/Interface/MueLu_HierarchyManager.hpp:96]: `numberOfMultiVectors_`?

**Comment by lucbv** [comment]
@jhux2 something seems to be going wrong with the reading of the fuego matrix when complex are enabled...

File: packages/muelu/src/Interface/MueLu_HierarchyManager.hpp (hunk 2 above)
Comment (jhux2) [reply packages/muelu/src/Interface/MueLu_HierarchyManager.hpp:96]: It's the number of individual vectors in a MultiVector.  I wasn't really happy with any of the names I came up with.  The current name is at least consistent with the usage/name in MueLu::Hierarchy.

Comment (cgcgcg) [reply packages/muelu/src/Interface/MueLu_HierarchyManager.hpp:96]: Yep, not sure either. Maybe `numberOfRHS`?

**Pull request merged**